modules/core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -1222,12 +1223,136 @@ object Decoder
     protected[this] final def formatMessage(input: String, message: String): String = message
   }
 
+  private[this] lazy val DurationPattern = Pattern.compile(
+    "([-+]?)P(?:([-+]?[0-9]+)D)?" + "(T(?:([-+]?[0-9]+)H)?(?:([-+]?[0-9]+)M)?(?:([-+]?[0-9]+)(?:[.,]([0-9]{0,9}))?S)?)?",
+    Pattern.CASE_INSENSITIVE
+  )
+
+  private[this] def parseDuration(text: CharSequence): Duration = {
+
+    /*
+     * Copyright (c) 2007-present, Stephen Colebourne & Michael Nascimento Santos
```

Comment:
So this is annoying ... either this, or we just don't test this on JDK 8, cause it's broken for reasons out of control /shrug.

Reply:
Its nice that we kinda fix a bug, for our JDK8 users with this, but ... for organizational purposes, can we move this code out of this file into its own class in a way we can remove it later without breaking bincompat?

Reply:
Eh, I just removed it and disabled the relevant tests on JDK8. I should have follow my gut instinct and left it alone.